### PR TITLE
Update analyzer version and fix new findings

### PR DESCRIPTION
**Pull request by buyaa-n**
Dogfooding the new analyzer `CA1854: Prefer the 'IDictionary.TryGetValue(TKey, out TValue)' method` that developed by @CollinAlpert. Fixing the findings

### CODE REVIEW

File: src/libraries/System.Data.OleDb/src/System.Data.OleDb.csproj [hunk 1]
```diff
@@ -1,9 +1,9 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent);$(NetCoreAppMinimum)-windows;$(NetCoreAppMinimum);netstandard2.0;$(NetFrameworkMinimum)</TargetFrameworks>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <!-- Supress CA2249: Consider using String.Contains instead of String.IndexOf to avoid ifdefs. -->
-    <NoWarn>$(NoWarn);CA2249</NoWarn>
+    <NoWarn>$(NoWarn);CA2249;CA1420</NoWarn>
```
Comment (buyaa-n) [inline src/libraries/System.Data.OleDb/src/System.Data.OleDb.csproj:6]: @jkoritzinsky I was not able to figure out how to fix `warning CA1420: Delegates with managed types as parameters or the return type require runtime marshalling to be enabled in the assembly where the delegate is defined.`. Please let me know how to handle this

21 warnings in total here and in `System.Drawing.Common`

File: src/libraries/Microsoft.Win32.SystemEvents/src/Microsoft/Win32/SystemEvents.cs [hunk 2]
```diff
@@ -1068,9 +1068,9 @@ private static void RemoveEventHandler(object key, Delegate? value)
 
             lock (s_eventLockObject)
             {
-                if (s_handlers != null && s_handlers.ContainsKey(key))
+                if (s_handlers != null && s_handlers.TryGetValue(key, out var val))
                 {
-                    List<SystemEventInvokeInfo> invokeItems = s_handlers[key];
+                    List<SystemEventInvokeInfo> invokeItems = val;
```
Comment (stephentoub) [inline src/libraries/Microsoft.Win32.SystemEvents/src/Microsoft/Win32/SystemEvents.cs:1073]: This extra temporary isn't necessary... you can delete this line and just rename `val` above to be `invokeItems`, e.g. `out List<SystemEventInvokeInfo>? invokeItems`.

File: src/libraries/System.Data.Odbc/src/Common/System/Data/Common/DbConnectionOptions.cs [hunk 3]
```diff
@@ -77,14 +77,14 @@ internal bool HasBlankPassword
...
                 {
-                    if (_parsetable.ContainsKey(KEY.Password))
+                    if (_parsetable.TryGetValue(KEY.Password, out var value))
                     {
-                        return string.IsNullOrEmpty(_parsetable[KEY.Password]);
+                        return string.IsNullOrEmpty(value);
                     }
                     else
-                    if (_parsetable.ContainsKey(SYNONYM.Pwd))
+                    if (_parsetable.TryGetValue(SYNONYM.Pwd, out var val))
```
Comment (stephentoub) [inline src/libraries/System.Data.Odbc/src/Common/System/Data/Common/DbConnectionOptions.cs:85]: Nit: can we move this line up to be `else if`?

File: src/libraries/Microsoft.XmlSerializer.Generator/src/Sgen.cs [hunk 4]
```diff
@@ -618,10 +618,8 @@ private static Assembly SgenAssemblyResolver(object source, ResolveEventArgs e)
                     return null;
                 }
 
-                if (s_referencedic.ContainsKey(assemblyname))
+                if (s_referencedic.TryGetValue(assemblyname, out var reference))
```
[the remark this answers is not in the document]
Comment (buyaa-n) [reply src/libraries/Microsoft.XmlSerializer.Generator/src/Sgen.cs:621]: Sure will update, it was just the default behavior of the fixer

[the remark this answers is not in the document]
Comment (buyaa-n) [reply src/libraries/Microsoft.XmlSerializer.Generator/src/Sgen.cs:621]: I should not have said `default` 😄, it is the only behavior, which I thought good enough, we could improve it to suggest the actual type, though not sure  if that will be doable for all the cases

File: src/libraries/System.Data.OleDb/src/System.Data.OleDb.csproj (hunk 1 above)
Comment (jkoritzinsky) [reply src/libraries/System.Data.OleDb/src/System.Data.OleDb.csproj:6]: My commits in https://github.com/dotnet/runtime/pull/69709 have the fixes for the new analyzer warnings.

Comment (buyaa-n) [reply src/libraries/System.Data.OleDb/src/System.Data.OleDb.csproj:6]: Thanks, will rebase after that merged

**Review by skundlony — APPROVED** [review]

File: src/libraries/Microsoft.XmlSerializer.Generator/src/Sgen.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (buyaa-n) [reply src/libraries/Microsoft.XmlSerializer.Generator/src/Sgen.cs]: Created [Improve suggested variable name and/or type of the fixer](https://github.com/dotnet/roslyn-analyzers/issues/6022) for further improvement

File: src/libraries/System.Data.OleDb/src/System.Data.OleDb.csproj [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (stephentoub) [inline src/libraries/System.Data.OleDb/src/System.Data.OleDb.csproj:1]: Revert this file?

**Review by stephentoub — APPROVED** [review]

**Pull request merged**